protocol-units/da/m1/light-node/src/v1/sequencer.rs

```diff
@@ -88,64 +97,77 @@ impl LightNodeV1 {
 
 	}
 
-	/// Attempts to submit a batch of blobs according to the chunk size
-	async fn submit_and_resize(&self, block_blobs: &mut Vec<celestia_types::Blob>, chunk_size : usize) -> Result<usize, anyhow::Error> {
-	
+	async fn submit_blocks(&self, blocks : &Vec<Block>) -> Result<(), anyhow::Error> {
 
-		// iter in chunks of 4
-		while !block_blobs.is_empty() {
-			let mut chunk: Vec<celestia_types::Blob> = Vec::with_capacity(chunk_size);
-			
-			for _ in 0..chunk_size {
-				if let Some(blob) = block_blobs.pop() {
-					chunk.push(blob);
-				} else {
-					break;
-				}
-			}
-			
-			match self.pass_through.submit_celestia_blobs(&chunk).await {
-				Ok(_) => {
-					info!(
-						target: "movement_timing",
-						batch_size = chunk.len(),
-						"submitted_blob_batch",
-					);
-				}
-				Err(e) => {
-					info!("failed to submit blocks: {:?}", e);
-					block_blobs.extend(chunk.into_iter());
-					return Ok(chunk_size - 1);
-				}
-			}
+		let uid = LOGGING_UID.load(Ordering::SeqCst);
+		info!(
+			target: "movement_timing",
+			batch_size = blocks.len(),
+			uid = uid, 
+			"inner_submitting_block_batch"
+		);
+		let mut block_blobs = Vec::new();
+		for block in blocks {
+			let block_blob = self.pass_through.create_new_celestia_blob(
+					bcs::to_bytes(block)
+					.map_err(|e| anyhow::anyhow!("Failed to serialize block: {}", e))?,
+			)?;
+			block_blobs.push(block_blob);
 		}
-	
+		for block in blocks {
+			info!(target: "movement_timing", block_id = %block.id(), "submitting_block");
+		}
+		self.pass_through.submit_celestia_blobs(&block_blobs).await?;
+		for block in blocks {
+			info!(target: "movement_timing", block_id = %block.id(), "submitted_block");
+		}
+		Ok(())
+	}
 
-		Ok(chunk_size)
+	pub async fn submit_with_heuristic(&self, blocks : Vec<Block>) -> Result<(), anyhow::Error> {
 
-	}
+		let mut heuristic : GroupingHeuristicStack<Block> = GroupingHeuristicStack::new(vec![
+			DropSuccess::boxed(),
+			ToApply::boxed(),
+			SkipFor::boxed(1, Splitting::boxed(2)),
```

Comment:
Since the `SkipFor` combinator is generic, the inner strategy does not need to be dynamically dispatched?
```suggestion
			SkipFor::boxed(1, Splitting::new(2)),
```

Reply:
Good call.

Reply:
Ah, actually, I think this starts forcing me down a path where I need to use associated types. Do you want me to do such a big rewrite?